<launch>
	<!-- Launch file created and tested on 191224 Nat -->
	<!-- Arguments -->
	<arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
	<arg name="configuration_basename" default="turtlebot3_lds_2d.lua"/>
	<arg name="cmd_vel_topic" default="/cmd_vel" />
	<arg name="odom_topic" default="odom" />
	<arg name="initial_pose_x" default="-1.0"/>
	<arg name="initial_pose_y" default="-1.0"/>
	<arg name="initial_pose_a" default="0.0"/>
    <arg name="use_gazebo" default="true" />
	<arg name="use_frontier_exploration" default="true" />
	<arg name="open_rviz" default="false"/>
	<arg name="save_path" default="na" />

	<!-- Gazebo -->
	<group if="$(arg use_gazebo)">
        <include file="$(find turtlebot3_gazebo)/launch/turtlebot3_oldrif.launch">
            <arg name="model" value="$(arg model)"/>
            <arg name="x_pos" value="$(arg initial_pose_x)"/>
            <arg name="y_pos" value="$(arg initial_pose_y)"/>
            <arg name="z_pos" value="0.0"/>
        </include>
    </group>
    
    <!-- Launching SLAM gmapping and rviz -->
    <group>
        <include file="$(find turtlebot3_slam)/launch/turtlebot3_slam.launch">
            <arg name="slam_methods" value="gmapping" />
			<arg name="maxUrange" value="1.5"/>
			<arg name="open_rviz" value="$(arg open_rviz)"/>
        </include>
    </group>


	<!-- move base -->
	<node pkg="move_base" type="move_base" name="move_base" respawn="true" output="screen" >
		<param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" />
		<rosparam file="$(find turtlebot3_navigation)/param/costmap_common_params_$(arg model).yaml" command="load" ns="global_costmap" />
		<rosparam file="$(find turtlebot3_navigation)/param/costmap_common_params_$(arg model).yaml" command="load" ns="local_costmap" />
		<rosparam file="$(find turtlebot3_navigation)/param/local_costmap_params.yaml" command="load" />
		<rosparam file="$(find turtlebot3_navigation)/param/global_costmap_params.yaml" command="load" />
		<rosparam file="$(find turtlebot3_navigation)/param/move_base_params.yaml" command="load" />
		<rosparam file="$(find turtlebot3_navigation)/param/dwa_local_planner_params_$(arg model).yaml" command="load" />
		<remap from="cmd_vel" to="$(arg cmd_vel_topic)"/>
		<remap from="odom" to="$(arg odom_topic)"/>
	</node>
	<group if="$(arg use_frontier_exploration)">
    	<!-- Frontier exploration -->
        <include file="$(find explore_lite)/launch/explore.launch" />
    </group>

	<!-- save map -->
	<node name="save_map" pkg="Nturtlebot3_nav" type="save_map1303.py" respawn="true" output="screen" >
		<param name="save_path" value="$(arg save_path)"/>
    </node>
</launch>